pytorch_lightning/root_module/root_module.py

```diff
@@ -113,6 +113,29 @@ def optimizer_step(self, epoch_nb, batch_nb, optimizer, optimizer_i, second_orde
         # clear gradients
         optimizer.zero_grad()
 
+    def tbptt_split_batch(self, batch, split_size):
+        """
+        Return list of batch splits. Each split will be passed to forward_step to enable truncated
+        back propagation through time. The default implementation splits all root level tensors at
+        dim=1 (i.e. time dim). It assumes that each tensor has the same time dim length.
+        :return:
+        """
+        time_dims = [x.shape[1] for x in batch if isinstance(x, torch.Tensor) and x.dims() > 1]
+        assert all(x == time_dims[0] for x in time_dims), "Tensor time dimension size is ambiguous"
+
+        splits = []
+        for t in range(0, time_dims[0], split_size):
+            batch_split = []
+            for i, x in enumerate(batch):
+                if isinstance(x, torch.Tensor):
```

Comment:
is  there a case when we still want to split non tensors? what happens,  for instance in decoding for a seq2seq ? wouldn't  we tbptt a few steps into decoding, so the targets might be python lists (ie, class labels?  word indexes, etc?)